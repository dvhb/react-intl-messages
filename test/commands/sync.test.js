const { test } = require('@oclif/test');
const { copyFile, mkdir, readJson, rmdir } = require('../utils');
const lokaliseKeysJson = require('./__mocks__/lokaliseKeys.json');

const messagesDir = 'messagesSync';

beforeAll(() => {
  mkdir(messagesDir);
  copyFile('__mock__/en.json', `${messagesDir}/_default.json`);
  copyFile('__mock__/en.json', `${messagesDir}/en.json`);
});

afterAll(() => {
  rmdir(messagesDir);
});

describe('sync lokalise', () => {
  const projectId = process.env.PROJECT_ID;
  const token = process.env.TOKEN;

  test
    .env({ PROJECT_ID: projectId, TOKEN: token })
    .nock('https://api.lokalise.co', api =>
      api
        .get(`/api2/projects/${projectId}/keys`)
        .query(true)
        .reply(200, { project_id: '139504615bd04772c3b220.60315670', keys: [] })
        .post(`/api2/projects/${projectId}/keys`, body => {
          expect(body).toMatchSnapshot();
          return true;
        })
        .reply(200, lokaliseKeysJson),
    )
    .stderr()
    .command(['sync', '--messagesDir', messagesDir, '--provider', 'lokalise'])
    .it('upload new message to lokalise', ctx => {
      expect(ctx.stderr).toBe('');
    });

  test
    .env({ PROJECT_ID: projectId, TOKEN: token })
    .nock('https://api.lokalise.co', api =>
      api
        .get(`/api2/projects/${projectId}/keys`)
        .query(true)
        .reply(200, {
          project_id: '139504615bd04772c3b220.60315670',
          keys: [{ key_name: { web: 'welcome' }, translations: [{ translation: 'Hello!', language_iso: 'en' }] }],
        }),
    )
    .stderr()
    .command(['sync', '--messagesDir', messagesDir, '--provider', 'lokalise'])
    .it('update existing message lokalise', ctx => {
      expect(readJson(`${messagesDir}/en.json`)).toMatchSnapshot();
      expect(ctx.stderr).toBe('');
    });
});

describe('sync locize', () => {
  const projectId = process.env.PROJECT_ID;
  const token = process.env.TOKEN;

  test
    .env({ PROJECT_ID: projectId, TOKEN: token })
    .nock('https://api.locize.io', api =>
      api
        .get(`/${projectId}/latest/en/test`)
        .reply(200, {})
        .post(`/missing/${projectId}/latest/en/test`, body => {
          expect(body).toMatchSnapshot();
          return true;
        })
        .reply(200, lokaliseKeysJson),
    )
    .stderr()
    .command(['sync', '--messagesDir', messagesDir, '--provider', 'locize'])
    .it('upload new message to locize', ctx => {
      expect(ctx.stderr).toBe('');
    });
  test
    .env({ PROJECT_ID: projectId, TOKEN: token })
    .nock('https://api.locize.io', api =>
      api
        .get(`/${projectId}/latest/en/test`)
        .reply(200, { welcome: { context: { text: 'Welcome message' }, value: 'Hello!' } }),
    )
    .stderr()
    .command(['sync', '--messagesDir', messagesDir, '--provider', 'locize'])
    .it('update local message from locize', ctx => {
      expect(readJson(`${messagesDir}/en.json`)).toMatchSnapshot();
      expect(ctx.stderr).toBe('');
    });
});
