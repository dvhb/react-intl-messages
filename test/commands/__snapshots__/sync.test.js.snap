// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sync locize update local message from locize 1`] = `
Array [
  Object {
    "defaultMessage": "Hello {name}, you have {unreadCount, number} {unreadCount, plural, one {message} other {messages}}",
    "description": "Welcome message",
    "files": Array [
      "src/App.js",
    ],
    "id": "welcome",
    "message": "Hello!",
  },
]
`;

exports[`sync locize update local message from locize: file/en.json 1`] = `
Array [
  Object {
    "defaultMessage": "Hello {name}, you have {unreadCount, number} {unreadCount, plural, one {message} other {messages}}",
    "description": "Welcome message",
    "files": Array [
      "src/App.js",
    ],
    "id": "welcome",
    "message": "Hello!",
  },
]
`;

exports[`sync locize upload new message to locize 1`] = `
Object {
  "welcome": Object {
    "context": Object {
      "text": "Welcome message",
    },
    "value": "Hello {name}, you have {unreadCount, number} {unreadCount, plural, one {message} other {messages}}",
  },
}
`;

exports[`sync locize upload new message to locize: en 1`] = `
Object {
  "welcome": Object {
    "context": Object {
      "text": "Welcome message",
    },
    "value": "Hello {name}, you have {unreadCount, number} {unreadCount, plural, one {message} other {messages}}",
  },
}
`;

exports[`sync locize upload new message to locize: ru 1`] = `
Object {
  "welcome": Object {
    "context": Object {
      "text": "Welcome message",
    },
    "value": "Привет!",
  },
}
`;

exports[`sync lokalise update existing message lokalise 1`] = `
Array [
  Object {
    "defaultMessage": "Hello {name}, you have {unreadCount, number} {unreadCount, plural, one {message} other {messages}}",
    "description": "Welcome message",
    "files": Array [
      "src/App.js",
    ],
    "id": "welcome",
    "message": "Hello!",
  },
]
`;

exports[`sync lokalise upload new message to lokalise 1`] = `
Object {
  "keys": Array [
    Object {
      "description": "Welcome message",
      "key_name": "welcome",
      "platforms": Array [
        "ios",
        "android",
        "web",
        "other",
      ],
      "translations": Array [
        Object {
          "language_iso": "en",
          "translation": "Hello {name}, you have {unreadCount, number} {unreadCount, plural, one {message} other {messages}}",
        },
      ],
    },
  ],
}
`;
